cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/SlowBrokerFinder.java

```diff
@@ -112,19 +113,26 @@ public SlowBrokerFinder() {
     // Preprocess raw metrics to get the derived metrics for each broker.
     Map<BrokerEntity, List<Double>> historicalValueByBroker = new HashMap<>();
     Map<BrokerEntity, Double> currentValueByBroker = new HashMap<>();
+    Set<Integer> skippedBrokers = new HashSet<>();
     for (Map.Entry<BrokerEntity, ValuesAndExtrapolations> entry : currentMetricsByBroker.entrySet()) {
       BrokerEntity entity = entry.getKey();
-      ValuesAndExtrapolations valuesAndExtrapolations = entry.getValue();
-      double latestTotalBytesIn = valuesAndExtrapolations.metricValues().valuesFor(ALL_TOPIC_BYTES_IN.id()).latest() +
-                                  valuesAndExtrapolations.metricValues().valuesFor(ALL_TOPIC_REPLICATION_BYTES_IN.id()).latest();
-      double latestLogFlushTime = valuesAndExtrapolations.metricValues().valuesFor(BROKER_LOG_FLUSH_TIME_MS_999TH.id()).latest();
+      AggregatedMetricValues aggregatedMetricValues = entry.getValue().metricValues();
+      // BROKER_LOG_FLUSH_TIME_MS_999TH comes from a Timer metric, which may not report if there is no value recorded (
+      // the broker serves no traffic and never performs log flush operation).
+      if (aggregatedMetricValues.valuesFor(BROKER_LOG_FLUSH_TIME_MS_999TH.id()) == null) {
```

Comment:
Is it possible to make this more generic by making this sanity check in a util function?  -- e.g. please see how we handle similar operation in `HolderUtils#allowMissingBrokerMetric(Cluster, int, RawMetricType)`.